easybuild/easyblocks/generic/buildenv.py

```diff
@@ -28,14 +28,52 @@
 
 @author: Alan O'Cais (Juelich Supercomputing Centre)
 """
+import os
+
 from easybuild.easyblocks.generic.bundle import Bundle
+from easybuild.tools.config import build_option, update_build_option
+from easybuild.tools.toolchain.toolchain import RPATH_WRAPPERS_SUBDIR
 
 
 class BuildEnv(Bundle):
     """
     Build environment of toolchain: only generate module file
     """
 
+    def prepare_step(self, *args, **kwargs):
+        """
+        Custom prepare step for buildenv: export rpath wrappers if they are being used
+        """
+        # We export the rpath wrappers under special conditions
+        # (the wrappers are needed if LD_LIBRARY_PATH is being filtered)
+        filtered_env_vars = build_option('filter_env_vars') or []
+        if build_option('rpath') and 'LD_LIBRARY_PATH' in filtered_env_vars and 'LIBRARY_PATH' not in filtered_env_vars:
+            # re-create installation dir (deletes old installation),
+            self.make_installdir()
+            # then set keeppreviousinstall to True (to avoid deleting wrappers we create)
+            self.cfg['keeppreviousinstall'] = True
+
+            # Temporarily unset the rpath setting so that we can control the rpath wrapper creation
+            update_build_option('rpath', False)
+
+            # Prepare the toolchain, we need to export the wrappers _after_ the modules have been loaded
+            # (so that correct compilers are defined)
+            super(BuildEnv, self).prepare_step(*args, **kwargs)
+
+            # export the rpath wrappers
+            self.toolchain.prepare_rpath_wrappers(
```

Comment:
Doing it this way is not good enough, it is not fully respecting the hooks. I will have to think of another approach that does the export as an integral part of the `prepare_step`.